extensions/rapids_notebook_files.py

```diff
@@ -15,24 +53,32 @@ def find_notebook_related_files(app, pagename, templatename, context, doctree):
     """
     if "examples/" in pagename and context["page_source_suffix"] == ".ipynb":
         source_root = pathlib.Path(__file__).parent / ".." / "source"
-        base_url = app.config.rapids_deployment_notebooks_base_url
+        output_root = pathlib.Path(app.builder.outdir)
         rel_page_parent = pathlib.Path(pagename).parent
         path_to_page_parent = source_root / rel_page_parent
+        path_to_output_parent = output_root / rel_page_parent
 
-        related_notebook_dirs = []
-        related_notebook_files = []
-        for page in path_to_page_parent.glob("*"):
-            if "ipynb" not in page.name:
-                url = f"{base_url}{rel_page_parent}/{page.name}"
-                if (path_to_page_parent / page).is_dir():
-                    related_notebook_dirs.append((page.name + "/", url))
-                else:
-                    related_notebook_files.append((page.name, url))
-
-        context["related_notebook_files"] = (
-            related_notebook_dirs + related_notebook_files
+        # Copy all related files to output and apply templating
+        related_notebook_files = walk_files(
+            app, path_to_page_parent, path_to_output_parent
         )
 
+        # Make archive of related files
+        if related_notebook_files and len(related_notebook_files) > 1:
+            archive_path = path_to_output_parent / "all_files.zip"
```

Comment:
nit: can we make the filename depend on the example? Maybe `pagename` contains the name of the example? A thing that makes creating a filename from it a bit tricky is that there are many characters that need dealing with when creating a filename, they somehow need to be encoded. So this might be something for a new PR?

Reply:
Yeah I started to implement this and it was less trivial than I thought so I backed out. Probably something for a follow-up PR as you say.